test/unit/app/models/test_models.py

```diff
@@ -249,6 +249,105 @@ def test_new_credit_transfer_rejected(create_credit_transfer):
+        and str(create_credit_transfer.transfer_type) in limit.applied_to_transfer_types
+    ]
+
+    # Check Sempo Level 2 LIMITS (payment only)
+    kyc = KycApplication(type='INDIVIDUAL')
+    kyc.user = create_credit_transfer.sender_user
+    kyc.kyc_status = 'VERIFIED'
+    assert create_credit_transfer.check_sender_transfer_limits() == [
+        limit for limit in LIMITS
+        if limit.name == 'Sempo Level 2'
+        and str(create_credit_transfer.transfer_type) in limit.applied_to_transfer_types
+    ]
+
+    # Check Sempo Level 3 LIMITS (payment only)
+    kyc.type = 'BUSINESS'
+    assert create_credit_transfer.check_sender_transfer_limits() == [
+        limit for limit in LIMITS
+        if limit.name == 'Sempo Level 3'
+        and str(create_credit_transfer.transfer_type) in limit.applied_to_transfer_types
+    ]
+
+    # Check GE LIMITS for Liquid Token (withdrawal only)
+    create_credit_transfer.token.token_type = token.TokenType.LIQUID
+    create_credit_transfer.transfer_type = "WITHDRAWAL"
+    create_credit_transfer.sender_transfer_account.balance = 10000
+    assert create_credit_transfer.check_sender_transfer_limits() == [
+        limit for limit in LIMITS
+        if limit.name == 'GE Liquid Token - Standard User'
+        and str(create_credit_transfer.transfer_type) in limit.applied_to_transfer_types
+    ]
+
+    # Check Limits skipped if no sender user (exchange)
+    create_credit_transfer.sender_user = None
+    assert create_credit_transfer.check_sender_transfer_limits() is None
+
+
+def test_new_credit_transfer_check_sender_transfer_limits_exception(external_reserve_token, create_credit_transfer):
+    """
+    GIVEN a CreditTransfer model
+    WHEN a new credit transfer is created
+    THEN check the correct check_sender_transfer_limits raises AccountLimitError
+    """
+    from server.utils.transaction_limits import LIMITS
+    from server.models import credit_transfer, token
+    from server import db
+
+    create_credit_transfer.token.token_type = token.TokenType.RESERVE
+    create_credit_transfer.sender_user.kyc_applications = []
+
+    # Sempo Level 0 LIMITS (payment only) on init
+    with pytest.raises(AccountLimitError):
+        c = credit_transfer.CreditTransfer(
+            amount=1000000,
+            token=external_reserve_token,
+            sender_user=create_credit_transfer.sender_user,
+            recipient_user=create_credit_transfer.sender_user,
+            transfer_type='PAYMENT'
+        )
+        db.session.add(c)
+        db.session.flush()
+
+    # Sempo Level 0 LIMITS (payment only) on check LIMITS
+    with pytest.raises(AccountLimitError):
+        create_credit_transfer.check_sender_transfer_limits()
+
+    # Check GE LIMITS for Liquid Token (withdrawal only) on check LIMITS
+    create_credit_transfer.token.token_type = token.TokenType.LIQUID
+    create_credit_transfer.transfer_type = "WITHDRAWAL"
+    create_credit_transfer.sender_transfer_account.balance = 1000
```

Comment:
would be good to test the % right on the boundary, and also to test transfer count. what if they did a transfer earlier this week vs two weeks ago

Reply:
ok